src/Robot-Dobot_MG400/main.py

```diff
@@ -195,10 +203,21 @@ def deinitialize(self):
 
     def configure(self):
 
+        self.movement_height = float(self.movement_height)
         self.set_acceleration_linear(self.acceleration_factor)  # Linear acceleration factor 1-100
         self.set_speed_global(self.global_speed_factor)  # Global speed factor 1-100
         self.set_speed_linear(self.speed_factor)  # Linear speed factor 1-100
-        self._last_xyzr = (None, None, None, None)
+        # self._last_xyzr = (None, None, None, None)
+        self._last_xyzr = self.get_position()
```

Comment:
I assume the purpose of this line is to ensure the robot does not move up and down if it already is at the correct position when the measurement starts?
Also assuming get_position is returning floats and the dobot is actually measuring the position (and not just returning the most recent set values), there will probably always be some deviation in the 0.001 range. Not necessarily an issue, just something to be aware of.

Reply:
When using the new jump mode the robot makes at the beginning a move to the movement height. Because move_linear always needs x, y, z, and r, it is necessary to read them out once at the beginning to be able to set the current x, y, and r when going to the movement height.